Fixed #30049 -- Fix GIS widget init after "Add another" in admin inlines

shaddeus opened the pull request:
This fixes a initialization of new GIS widget that was added using "Add another" button in admin inlines. The "Add another" button takes a hidden `__prefix__`-ed form, copies it, updates element attributes IDs and NAMEs and put down at right place. The GIS widget template uses inline javascript that initializes widget using IDs and that was a problem!

Proposal solution adds hook for "Add another" action using EventListener for `formset:added` javascript event and fixes inline javascript code and refires it.

claudep commented:
Thanks for the patch, but please avoid introducing a jQuery dependency in that file. We strive to write as much pure Javascript as possible.

shaddeus commented:
Thanks for reply! Well, I'll rewrite it into pure Javascript and I'll try to kickout a call of `eval()` function. I'll have a next free day for this in one or two months. I'll fix it!

shaddeus commented:
@claudep I tried vanilla javascript. I cannot catch jQuery event [`formset:added`](https://github.com/django/django/blob/master/django/contrib/admin/static/admin/js/inlines.js#L126) from `inlines.js` with vanilla javascript. It's impossible due jQuery has own event system build on top of the DOM event system.
https://stackoverflow.com/questions/25256173/can-i-use-jquery-trigger-with-event-listeners-added-with-addeventlistener

May I update `inlines.js` and rewrite the event `formset:added` trigger using vanilla javascript? Send with jquery object.
Or can I use jQuery in `OLMapWidget.js` or do you have some idea how to deal with this problem?

claudep commented:
If feasible, I would prefer changing the event trigger in `inline.js`. Could you explore that solution?

felixxm commented:
Closing due to inactivity.
